janusgraph-backend-testutils/src/main/java/org/janusgraph/graphdb/JanusGraphTest.java

```diff
@@ -5908,6 +5908,88 @@ public void testGraphCentricQueryProfiling() {
         assertEquals(fullScanAnnotations, nested.getAnnotations());
     }
 
+    @Test
+    public void testGraphCentricQueryProfilingWithLimitAdjusting() {
+        Runnable dataLoader = () -> {
+            final PropertyKey name = makeKey("name", String.class);
+            final JanusGraphIndex compositeNameIndex = mgmt.buildIndex("nameIdx", Vertex.class).addKey(name).buildCompositeIndex();
+            finishSchema();
+
+            newTx();
+            for (int i = 0; i < 3000; i++) {
+                tx.addVertex("name", "bob");
+            }
+            tx.commit();
+        };
+
+        clopen(option(ADJUST_LIMIT), false);
+        dataLoader.run();
+
+        newTx();
+        Metrics mCompSingle = tx.traversal().V().has("name", "bob").profile().next().getMetrics(0);
```

Comment:
It looks like this:
```
Traversal Metrics
Step                                                               Count  Traversers       Time (ms)    % Dur
=============================================================================================================
JanusGraphStep([],[name.eq(bob)])                                   3000        3000          23.487   100.00
  constructGraphCentricQuery                                                                   0.035
  constructGraphCentricQuery                                                                   1.311
  GraphCentricQuery                                                                           23.842
    \_condition=(name = bob)
    \_orders=[]
    \_isFitted=true
    \_isOrdered=true
    \_query=multiKSQ[1]@4000
    \_index=nameIdx
    backend-query                                                   1000                       7.681
    \_query=nameIdx:multiKSQ[1]@1000
    \_limit=1000
    backend-query                                                   2000                       8.124
    \_query=nameIdx:multiKSQ[1]@2000
    \_limit=2000
    backend-query                                                   3000                       7.708
    \_query=nameIdx:multiKSQ[1]@4000
    \_limit=4000
                                            >TOTAL                     -           -          23.487        -
```

Reply:
But only in the case below, with `ADJUST_LIMIT=true`, right?

Reply:
Yes, thanks for the reminder. The comment above was posted at the wrong place. It belongs to the case below, i.e. with `ADJUST_LIMIT = true`.

Here the profiling output should be:
```
Traversal Metrics
Step                                                               Count  Traversers       Time (ms)    % Dur
=============================================================================================================
JanusGraphStep([],[name.eq(bob)])                                   3000        3000         123.323   100.00
  constructGraphCentricQuery                                                                   0.086
  constructGraphCentricQuery                                                                   1.073
  GraphCentricQuery                                                                          121.164
    \_condition=(name = bob)
    \_orders=[]
    \_isFitted=true
    \_isOrdered=true
    \_query=multiKSQ[1]@100000
    \_index=nameIdx
    backend-query                                                   3000                     120.870
    \_query=nameIdx:multiKSQ[1]@100000
    \_limit=100000
                                            >TOTAL                     -           -         123.323        -
```